bears/vcs/git/GitCommitBear.py

```diff
@@ -203,3 +240,71 @@ def check_body(self, body,
                                'characters.'.format(body_line_length))
+
+    def check_issue_reference(self, body,
+                              body_close_issue: bool=False,
+                              body_close_issue_full_url: bool=False,
+                              body_close_issue_on_last_line: bool=False):
+        """
+        Check for matching issue related references and URLs.
+
+        :param body:
+            Body of the commit message of HEAD.
+        :param body_close_issue:
+            Whether to check for the presence of issue close reference
+            within the commit body by retrieving host information from git
+            configuration. GitHub and GitLab support auto closing issues with
+            commit messages. Checks for matching keywords in the commit body.
+            By default, if none of ``body_close_issue_full_url`` and
+            ``body_close_issue_on_last_line`` are enabled, checks for presence
+            of short references like ``closes #213``. Otherwise behaves
+            according to other chosen flags.
+            More on keywords follows.
+            [GitHub](https://help.github.com/articles/closing-issues-via-commit-messages/)
+            [GitLab](https://docs.gitlab.com/ce/user/project/issues/automatic_issue_closing.html)
+        :param body_close_issue_full_url:
+            Checks the presence of issue close reference with a full URL
+            related to some issue. Works along with ``body_close_issue``.
+        :param body_close_issue_on_last_line:
+            When enabled, checks for issue close reference presence on the
+            last line of the commit body. Works along with
+            ``body_close_issue``.
+        """
+        if not body_close_issue:
+            return
+
+        host = self.get_host_from_remotes()
+        if host not in self.SUPPORTED_HOSTS:
+            return
+
+        if body_close_issue_on_last_line:
+            body = body.splitlines()[-1]
+            result_message = ('Body of HEAD commit does not contain any {} '
+                              'reference in the last line.')
+        else:
+            result_message = ('Body of HEAD commit does not contain any {} '
+                              'reference.')
+
+        if body_close_issue_full_url:
+            result_message = result_message.format('full issue')
+            initial_check_regex = r'https?://{}\S+/issues/\w+'.format(host)
+            final_check_regex = r'https?://\S+[1-9][0-9]*'
+        else:
+            result_message = result_message.format('issue')
+            initial_check_regex = r'#\w+'
+            final_check_regex = r'(#[1-9][0-9]*)'
+
+        join_regex = r'({0}(?:\s*(?:,|and)\s*{0})*)'
+        matches = re.findall(self.SUPPORTED_HOSTS[host]
+                             + join_regex.format(initial_check_regex), body)
+        matches = [re.findall(initial_check_regex, match) for match in matches]
```

Comment:
with `re.split` you get the list you want :) actually you don't need to flatten everything, just use a for-loop directly:
```
for issue_match in ...:
    for issue_id in re.split(issue_match, ...):
        pass
```

>> huge modification to all regexes
>
> Do you mean to say that we could combine the check for both of them with just one regular 
> expression?

yes :+1:

> BTW, initial_check_regex is different for both cases, since I'm verifying URL right in itself in the 
> first case for long URLs.
> This is crucial for proper error reporting in issue numbers and wrong URLs.